app/src/androidTest/java/com/github/se/travelpouch/di/TestModule.kt

```diff
@@ -18,19 +38,80 @@ object TestModule {
 
   @Provides
   @Singleton
-  fun provideFirebaseAuth(): AuthenticationService {
-    return MockFirebaseAuthenticationService()
+  fun provideFirebaseAuth(): FirebaseAuth {
+    return Firebase.auth.apply { useEmulator("10.0.2.2", 9099) }
   }
 
   @Provides
   @Singleton
-  fun providesProfileRepository(): ProfileRepository {
-    return ProfileRepositoryMock()
+  fun provideFirebaseFirestore(): FirebaseFirestore {
+    return Firebase.firestore.apply { useEmulator("10.0.2.2", 8080) }
   }
 
   @Provides
   @Singleton
-  fun providesTravelRepository(): TravelRepository {
-    return TravelRepositoryMock()
+  fun provideFirebaseStorage(): FirebaseStorage {
+    val storageBucket =
+        FirebaseApp.getInstance().options.storageBucket
+            ?: throw IllegalStateException(
+                "Firebase storage bucket not set in google-services.json")
+    return Firebase.storage("gs://$storageBucket").apply { useEmulator("10.0.2.2", 9199) }
```

Comment:
I think it would be interesting to have a log or error handling if useEmulator fail. Is this possible and necessary ?

Reply:
This is not necessary in my opinion. It will fail if the storage bucket is not defined in the google-services.json. Which would be an issue and prevent the tests to start normally. For the logging, the error seems clear enough to me.
Do you agree ?